src/core/stream/buffers_handler.ts

```diff
@@ -403,9 +413,25 @@ export default function BuffersHandler(
             })
         );
 
+    const lastPeriodBuffer$ = currentBuffer$
+      .filter((evt) => {
+        return evt.type === "periodBufferReady" || evt.type === "representationChange";
+      })
```

Comment:
So this is still the last created buffer, not the last Period?

From what I can see for now, the logic should probably be:
  - ok we received a full Event
  - are we the last period?
yes -> Indicate that the last segment has been considered (but not pushed yet, so there's another problem there before calling ``endOfStream``)
no -> create the next buffer